order-service/src/controllers/order.controller.ts

```diff
@@ -16,36 +19,45 @@ export class OrderController {
             return;
         }
 
-        if (isNaN(userId)) {
-            res.status(HttpStatus.BAD_REQUEST).json(new ServerError(
-                Messages.validation.id_must_be_number("userId"),
-                ErrorCode.NOT_ENOUGH_DATA,
-            ).toJSON());
-        }
+        for (let param of numberParams) {
+            if (isNaN(param)) {
+                res.status(HttpStatus.BAD_REQUEST).json(new ServerError(
+                    Messages.validation.id_must_be_number(param),
+                    ErrorCode.NOT_ENOUGH_DATA,
+                ).toJSON());
 
-        if (isNaN(placeId)) {
-            res.status(HttpStatus.BAD_REQUEST).json(new ServerError(
-                Messages.validation.id_must_be_number("placeId"),
-                ErrorCode.NOT_ENOUGH_DATA,
-            ).toJSON());
+                return;
+            }
         }
 
-        if (isNaN(checkoutSlotId)) {
+        if (items.length) {
```

Comment:
You could also validate if items is not a string. As it is, it will display item_data_needs_to_be_provided instead of items_must_be_an_array.

Reply:
Great! Done